core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,643 @@
+      if (!(hasExternalElementReference)) {
+        deleteList.add(annotation);
+      }
+    }
+    deleteList = deleteList.stream().filter(x -> Objects.nonNull(x.getDetails().get(IS_ELEMENT_TITLE_BLOCK)))
+        .filter(x -> x.getDetails().get(IS_ELEMENT_TITLE_BLOCK).equals(TRUE)).collect(Collectors.toList());
+    CapellaServices.getService().removeElements(deleteList);
+    for (DAnnotation annotation : deleteList) {
+      clearEAnnotations(elementView, annotation);
+    }
+  }
+
+  public void clearEAnnotations(Object elementView, DAnnotation element) {
+    List<DAnnotation> annotationsList = new ArrayList<>();
+    for (EObject titleBlockLine : element.getReferences()) {
+      if (titleBlockLine instanceof DAnnotation) {
+        annotationsList.add((DAnnotation) titleBlockLine); // title block lines
+        if (!((DAnnotation) titleBlockLine).getReferences().isEmpty()) {
+          for (EObject titleBlockCell : ((DAnnotation) titleBlockLine).getReferences()) {
+            if (titleBlockCell instanceof DAnnotation) {
+              annotationsList.add((DAnnotation) titleBlockCell);
+            }
+          }
+        }
+      }
+    }
+    ((DSemanticDiagram) elementView).getEAnnotations().removeAll(annotationsList);
+  }
+
+  public void clearLineEAnnotation(DAnnotation titleBlock, DDiagram diagram) {
+    List<DAnnotation> eAnnotationsList = diagram.getEAnnotations();
+    List<DAnnotation> annotationsListToBeRemoved = new ArrayList<>();
+    for (DAnnotation eAnnotation : eAnnotationsList) {
+      if (eAnnotation.getReferences().contains(titleBlock)) {
+        annotationsListToBeRemoved.add(eAnnotation);
+        for (EObject reference : eAnnotation.getReferences()) {
+          if (reference instanceof DAnnotation) {
+            annotationsListToBeRemoved.add((DAnnotation) reference);
+          }
+        }
+      }
+    }
+    eAnnotationsList.removeAll(annotationsListToBeRemoved);
+  }
+
+  public void clearColumnEAnnotation(DAnnotation titleBlock, DDiagram diagram) {
+    String columnNumber = titleBlock.getDetails().get(COLUMN);
+    List<DAnnotation> eAnnotationsList = diagram.getEAnnotations();
+    List<DAnnotation> annotationsListToBeRemoved = new ArrayList<>();
+    DAnnotation parentTitleBlock = getParentTitleBlock(titleBlock, diagram);
+    for (EObject reference : parentTitleBlock.getReferences()) {
+      if (reference instanceof DAnnotation) {
+        List<EObject> lineReferences = ((DAnnotation) reference).getReferences();
+        for (EObject lineRef : lineReferences) {
+          if (lineRef instanceof DAnnotation) {
+            if (((DAnnotation) lineRef).getDetails().get(COLUMN).equals(columnNumber)) {
+              annotationsListToBeRemoved.add((DAnnotation) lineRef);
+            }
+          }
+        }
+      }
+    }
+    eAnnotationsList.removeAll(annotationsListToBeRemoved);
+  }
+
+  public List<Object> getTitleBlockCellContent(EObject diagram, EObject cell, EObject containerView) {

```

Comment:
There are important points to this approach:
1. Do we need to temporary TB_CONTENT annotation to store the result of the queries? Could you try to use the TitleBlockCell itself as the domain object of the node and inteprete the content on the fly?
2. Not only Strings are stored in these DT_TitleBlockCellNode. We have to support the query that returns a list of object. For example: feature:ownedPropertyValues. This is the basic use case and must be implemented.

Reply:
2. Implemented.

Reply:
1. TB_CONTENT shall hold the query (aql:, feature:, capella:), the result is interpreted on the fly.